source/daplink/error.c

```diff
@@ -117,6 +117,100 @@ static const char *const error_message[] = {
     "The bootloader CRC did not pass.",
 
 };
+
+static const char *const error_type[] = {
```

Comment:
Some of these errors can be categorized as multiple types. This may need to be a bitmask of errors.